File: math/number-theory/approximate_equal.rs
// Copyright https://github.com/microwind
// Approximate Equality Check

fn approx_equal(a: f64, b: f64, epsilon: f64) -> bool {
    (a - b).abs() < epsilon
}

fn main() {
    println!("Approximate Equality Check:");
    println!("0.1 + 0.2 ≈ 0.3: {}", approx_equal(0.1 + 0.2, 0.3, 1e-10));
}

File: math/number-theory/ApproximateEqual.ts
/**
 * Copyright https://github.com/microwind
 * Approximate Equality Check
 */

function approxEqual(a: number, b: number, epsilon: number): boolean {
    return Math.abs(a - b) < epsilon;
}

function main(): void {
    console.log("Approximate Equality Check:");
    console.log("0.1 + 0.2 ≈ 0.3:", approxEqual(0.1 + 0.2, 0.3, 1e-10));
}

main();
export { approxEqual };
